shared/crypto/ed25519.go

```diff
@@ -195,6 +195,10 @@ func GenerateAddress() (Address, error) {
 	return pk.Address(), nil
 }
 
+func (pub Ed25519PublicKey) MarshalJSON() ([]byte, error) {
```

Comment:
This would work if `encKeyPair` had public key as a pointer receiver type. https://github.com/0xBigBoss/pocket/blob/147fd72e1516d813e4bfd21c14c4504508b5fac8/shared/crypto/keypair.go#L46

```go
// encKeyPair struct stores the public key and the passphrase encrypted private key
// The encrypted private key is stored as a JSON string with the fields needed to decrypt the key
type encKeyPair struct {
	PublicKey     PublicKey `json:"public_key"`
	PrivKeyArmour string    `json:"priv_key_armour"`
}
```